src/App.tsx

```diff
@@ -1,7 +1,47 @@
 import React from 'react';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <p className="App__message">The last pressed key is [Enter]</p>
-  </div>
-);
+type State = {
+  pressedKey: string,
+};
+
+export class App extends React.Component<{}, State> {
+  state: Readonly<State> = {
+    pressedKey: '',
+  };
+
+  componentDidMount() {
+    document.addEventListener('keyup', this.handleKeyup);
+  }
+
+  handleKeyup = (event: KeyboardEvent) => {
+    let keyName = event.key;
+
+    if (keyName === ' ') {
+      keyName = 'space';
+    }
+
+    this.setState({ pressedKey: keyName });
+  };
+
+  commentWillUnmount() {
+    document.removeEventListener('keyup', this.handleKeyup);
+  }
+
+  render() {
+    const { pressedKey } = this.state;
+
+    return (
+      <div className="App">
+        {pressedKey ? (
+          <p className="App__message">
+            {`The last pressed key is [${pressedKey}]`}
+          </p>
+        ) : (
+          <p className="App__message">
+            Nothing was pressed yet
+          </p>
```

Comment:
fix style for ternar operator
![image](https://user-images.githubusercontent.com/113928955/209984691-6cc03665-7827-416b-b9ad-27423ee36676.png)